sdk/tables/Azure.Data.Tables/src/QueryOptions.cs

```diff
@@ -12,5 +12,7 @@ internal partial class QueryOptions
         public string NextPartitionKey { get; set; }
         /// <summary> The row key continuation token for an Entity request. </summary>
         public string NextRowKey { get; set; }
+        /// <summary> The If-Match header value for an Entity request. </summary>
+        public string ETag { get; set; }
```

Comment:
Use ETag type

Reply:
Tables uses weak etags, which the ETag type doesn't allow:
https://github.com/Azure/azure-sdk-for-net/blob/965869fd80da2487ad08389a8cc6172f6d63d68d/sdk/core/Azure.Core/src/ETag.cs#L86

Reply:
Time to implement https://github.com/Azure/azure-sdk-for-net/issues/9501 :)

I'm fine if we do it as a followup